### PR TITLE
Updated Date.parse()

### PR DESCRIPTION
**Pull request by dletorey**
### Description

- Added a note about Daylight Saving and how it can affect parsed Dates with no time zone set
- Updated the Non-Standard date strings

### Motivation

Working on the updates for [`Date.parse()` in Firefox 123](https://github.com/mdn/content/issues/319140)

### Additional details

- [https://bugzilla.mozilla.org/show_bug.cgi?id=1870434](https://bugzilla.mozilla.org/show_bug.cgi?id=1870434)
- [https://bugzilla.mozilla.org/show_bug.cgi?id=1870570](https://bugzilla.mozilla.org/show_bug.cgi?id=1870570)
- [https://bugzilla.mozilla.org/show_bug.cgi?id=1872333](https://bugzilla.mozilla.org/show_bug.cgi?id=1872333)

### Related issues and pull requests

- No BCD needed
- [Firefox Release PR](https://github.com/mdn/content/pull/32257)

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/date/parse/index.md [hunk 1]
```diff
@@ -58,6 +58,8 @@ Date.parse("2019-01-01T00:00:00");
 
 Implementations usually default to the local time zone when the date string is non-standard. For consistency, we will assume that the code uses the UTC timezone.
 
+> **Note:** The local time zone comes from the system setting of the device, for example _London/Europe_ GMT. Daylight saving affects this, which is set by the local Government to that time zone, this can affect the parsed date. In the UK they experiment with no DST from 18 Feb 1968 02:00 to 31 Oct 1971 03:00 which set the local time to UTC +1. This means that `Date.parse()` between those dates will be out by minus 1 hour or `-360000` milliseconds. This is not specific to the UK most countries have [experimented with DST for different amounts of time](https://en.wikipedia.org/wiki/Daylight_saving_time_by_country#Past_observance). Added the specific time zone will correct this `Date.parse("Jan 1, 1970 GMT")`.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/date/parse/index.md:61]: This paragraph is hard to parse for me. It is easier to just point out that the timezone offset (relative to UTC) is calculated based on the timezone at the location of the device, and at the time denoted by the time itself, not at the time the code is run. You can then link to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTimezoneOffset, which does a much better exposition. (The London example is very interesting—maybe add it to that page?) However, none of this has anything special to do with `Date.parse`, so it's certainly out of context here, unless you add it to every method that deals with local timezones.

File: files/en-us/web/javascript/reference/global_objects/date/parse/index.md [hunk 2]
```diff
@@ -75,12 +77,14 @@ Date.parse("Thu, 01 Jan 1970 00:00:00 GMT+0300");
 
 // Single number
 Date.parse("0");
-// 946684800000 in Chrome (Sat Jan 01 2000 00:00:00 GMT+0000);
-// NaN in Firefox;
+// Nan in Firefox ≤122
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/date/parse/index.md:80]:
```suggestion
// NaN in Firefox ≤122
```

**Review by Josh-Cena — CHANGES_REQUESTED** [review]

File: files/en-us/web/javascript/reference/global_objects/date/parse/index.md (hunk 1 above)
Comment (dletorey) [reply files/en-us/web/javascript/reference/global_objects/date/parse/index.md:61]: Cool, I shall get back on that next week (going on holiday in 10 mins).

BTW it is not just a London this it is most countries especially in Europe and at different times I can across this as i called `Date.parse("1 Jan 1970")` and it returned `-360000` doing further research I found it was everywhere.

**Review by pepelsbey — APPROVED** [review]
LGTM, thank you!
